File: public/js/tumblr/controllers/TumblrStatsController.js
'use strict';

/**
 * Tumblr Stats Controller.
 */
function TumblrStatsController($scope, $http) {

    $scope.nbDocuments = 0;

    $http.get("api/tumblr/stats").success(function(data) {
        $scope.nbDocuments = data.count;
    });
}

File: app/tumblr/controllers/TumblrStatsController.scala
package tumblr.controllers

import scala.concurrent.ExecutionContext
import ExecutionContext.Implicits.global

import play.api.libs.json._
import play.api.mvc._

import tumblr.services.SiteStatsService

object TumblrStatsController extends Controller {

  def index = Action.async {
    SiteStatsService.generate.map(stats => {
      Ok(Json.toJson(stats)).as("application/json")
    }).recover { case e => BadRequest(e.getMessage) }
  }

}
